CorsixTH/Lua/dialogs/edit_room.lua

```diff
@@ -1072,31 +1072,92 @@ function UIEditRoom:draw(canvas, ...)
   UIPlaceObjects.draw(self, canvas, ...)
 end
 
+--25 north wall, west
+--26 north wall, east
+--27 east wall, south
+--28 east wall, north
+--29 south wall, west
+--30 south wall, east
+--31 west wall, south
+--32 west wall, north
+-- single door blue print values matching TH
+local door_floor_blueprint_markers = {
+  north = 26,
+  east = 27,
+  south = 30,
+  west = 31
+}
+
+local window_floor_blueprint_markers = {
+  north = 33,
+  east = 34,
+  south = 35,
+  west = 36,
+}
+
+
+function UIEditRoom:onLeftButtonDown(x, y)
+  if self.phase == "walls" then
+    if 0 <= x and x < self.width and 0 <= y and y < self.height then -- luacheck: ignore 542
+    else
+      local mouse_x, mouse_y = self.ui:ScreenToWorld(self.x + x, self.y + y)
+      self.mouse_down_x = math.floor(mouse_x)
+      self.mouse_down_y = math.floor(mouse_y)
+      if self.move_rect then
+        self.move_rect_x = self.mouse_down_x - self.blueprint_rect.x
+        self.move_rect_y = self.mouse_down_y - self.blueprint_rect.y
+      elseif not self.resize_rect then
+        self:setBlueprintRect(self.mouse_down_x, self.mouse_down_y, 1, 1)
+      end
+    end
+  elseif self.phase == "door" then
+    if self.blueprint_door.valid then
+      self.ui:playSound("buildclk.wav")
+      self:confirm(true)
+    else
+      self.ui:tutorialStep(3, 9, 10)
+    end
+  elseif self.phase == "windows" then
+    self:placeWindowBlueprint()
+  end
+end
+
+-- Remove window
```

Comment:
Sorry, I should've helped with doc on this one.

```suggestion
--! Attempt to remove a window placed on the room blueprints
--!param x co-ordinate of target tile
--!param y co-ordinate of target tile
```